Fix Plan.create

therefromhere opened the pull request:
Allows Plan.create to accept product as any of a string id, dj-stripe object, or stripe dict.

* Use sync_from_stripe_data in Plan.create() for consistency
* Make StripeModel._api_create() convert any djstripe object kwargs to id
* Added test of Plan.create() with product as id, dj-stripe object, or stripe dict

Resolves #870

therefromhere committed:
Fix Plan.create

* Use sync_from_stripe_data in Plan.create() for consistency
* Make StripeModel._api_create() convert any djstripe object kwargs to id
* Added test of Plan.create() with product as id, dj-stripe object, or stripe dict

Resolves #870

kavdev commented on djstripe/models/base.py:
```diff
@@ -109,6 +109,10 @@ def _api_create(cls, api_key=djstripe_settings.STRIPE_SECRET_KEY, **kwargs):
+		for key in kwargs:
+			if isinstance(kwargs[key], StripeModel):
+				kwargs[key] = kwargs[key].id
+
```
There's another way to handle this piece. Idk if it's still in the codebase, but check out how we handle passing plan ids to a subscription. I'm currently on mobile, but I can gather example links for you next time I get a free moment.

therefromhere replied:
Ah right, like this?

https://github.com/dj-stripe/dj-stripe/blob/0580d6f2511cd990897d1bba5155dd2730ec6f1c/djstripe/models/billing.py#L1038-L1040

kavdev replied:
exactly

therefromhere committed:
Convert Product to id only in Plan.create

instead in for all StripeModel objects in StripeModel._api_create
As requested by kavdev in review
